Kernel_23/include/CGAL/Kernel/function_objects.h

```diff
@@ -626,9 +626,35 @@ class Compare_weighted_squared_radius_3
 };
 
 
+  template <typename K>
+  class Compare_slope_3
+  {
+    typedef typename K::FT                 FT;
+    typedef typename K::Point_3 Point_3;
+  public:
+    typedef typename K::Comparison_result  result_type;
+
+    result_type operator()(const Point_3& p, const Point_3& q, const Point_3& r, const Point_3& s) const
+    { 
+      Comparison_result sign_pq = compare(q.z(),p.z());
+      Comparison_result sign_rs = compare(s.z(),r.z());
+      
+      if(sign_pq != sign_rs){
+        return compare(static_cast<int>(sign_pq), static_cast<int>(sign_rs));
+      }
 
+      if((sign_pq == EQUAL) && (sign_rs == EQUAL)){
+        return EQUAL;
+      }
 
-  //////////////////////
+      CGAL_assertion( (sign_pq == sign_rs) && (sign_pq != EQUAL)  );
+      
+      Comparison_result res = CGAL::compare(square(p.z() - q.z()) * squared_distance(r,s),
+                                            square(r.z() - s.z()) * squared_distance(p,q));
+      return (sign_pq == SMALLER) ? opposite(res) : res;
```

Comment:
As suggested by @mglisse in the small feature, instead of `squared_distance`, to save a multiplication and an addition, we could use the squared distance in the projection on the 2D plane: `(square(r.x())+square(s.x()))`. That does not change the result: with the current implementation we compare two sines, and with the more efficient version we would compare two tangents. And both `sin()` and `tan()` are increasing between -90° and 90°.